### DISCUSSION TITLE
APIMAN authentication

### DISCUSSION THREAD

**Discussion by cleuson-alves** [apiman/apiman#2611] [General]
Hi, reading the apiman documentation (https://www.apiman.io/apiman-docs/installation-guide/latest/servlet/wildfly.html), I saw that it is necessary to use keycloak for authentication, I would like to know if it is available another form of authentication like Google, LDAP etc, or do you absolutely need to use just keycloak? Thanks

**Comment by msavy** [comment]
Hi @cleuson-alves, glad you are enjoying Apiman.

My answer here assumes you are specifically talking about logging into the Apiman Manager UI (web interface):

There are several options: 
1. We are developing additional integrations with my company Black Parrot Labs [in the downstream](https://www.blackparrotlabs.io/black-parrot-labs-enterprise). Working with us supports continued development of the project.
1. Develop your own integration; the project is open source and this should be feasible (and feel free to contribute).
1. Use Keycloak's social logins and federation, which allows you to authenticate via LDAP, social logins, etc.
1. Wait for Apiman 4, where this it should be easier to get other auth providers working (although again, we will likely be developing more specific and well-tested support this as part of Black Parrot Labs Enterprise Suite)

I hope that answers your question!
